algorithms/src/polycommit/kzg10/degree_info.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::{ensure, Result};
+use std::collections::HashSet;
+
+#[derive(Clone)]
+pub struct DegreeInfo {
+    /// max degree of the required SRS to commit to the polynomials
+    pub max_degree: usize,
+    /// max IOP poly degree used for (i)fft_precomputation
+    pub max_fft_size: usize,
+    /// degree bounds on IOP polynomials
+    pub degree_bounds: Option<HashSet<usize>>,
+    /// hiding bound for polynomial queries
+    pub hiding_bound: usize,
+    /// supported lagrange-basis SRS
+    pub lagrange_sizes: Option<HashSet<usize>>,
+}
+
+impl DegreeInfo {
+    pub fn union(self, other: &Self) -> Result<Self> {
+        ensure!(self.hiding_bound == other.hiding_bound);
```

Comment:
We don't need this `ensure`, right? The union should just return the maximum of the two `hiding_bound`s. If we do that, then the output of this method can be a `Self`, which will clean up a bunch of other code.